core/controllers/reader_test.py

```diff
@@ -205,6 +257,106 @@ def test_huge_answers(self):
             'a' * 1000500, 'Sorry, nope, we didn\'t get it', '')
 
 
+class ReaderClassifyTests(test_utils.GenericTestBase):
+    """Test reader.classify using the sample explorations.
+
+    Since the end to end tests cover correct classification,
+    ReaderClassifyTests is only checking which of the hard/soft/classifier
+    rules is classified on input.
+    """
+
+    def setUp(self):
+        super(ReaderClassifyTests, self).setUp()
+        self._init_classify_inputs('16')
+
+    def _init_classify_inputs(self, exploration_id):
+        test_exp_filepath = os.path.join(
+            feconf.TESTS_DATA_DIR, 'string_classifier_test.yaml')
+        yaml_content = utils.get_file_contents(test_exp_filepath)
+        assets_list = []
+        exp_services.save_new_exploration_from_yaml_and_assets(
+            feconf.SYSTEM_COMMITTER_ID, yaml_content,
+            'Testing String Classifier', 'Test',
+            exploration_id, assets_list)
+
+        self.EXP_ID = exploration_id
+        self.EXP_STATE = (
+            exp_services.get_exploration_by_id(exploration_id).states['Home'])
+
+    def _get_classiying_rule_type(self, answer):
+        response = reader.classify(
+            self.EXP_ID, self.EXP_STATE, answer, {'answer': answer})
+        return response['matched_rule_type']
+
+    def test_hard_rule_classification(self):
```

Comment:
Sure, so my interpretation is to run a similar test except with less data for ease of understanding. Personally I think more data enables better understanding of expected behavior compared to hindering it, but for this specific case I don't mind as much. Except when I look through core.tests.test_utils I can't find create_new_default_exploration or create_new_valid_exploration.

I planned to book some time to chat on Wednesday anyways as you seem to have the most time then. Since we've been going back and forth on this let's discuss it then.